crates/viewer/re_dataframe_ui/src/filters/boolean_filter.rs

```diff
@@ -0,0 +1,144 @@
+use arrow::datatypes::{DataType, Field};
+use datafusion::common::Column;
+use datafusion::logical_expr::{Expr, col, lit};
+use datafusion::prelude::{array_element, array_has, array_sort};
+use re_ui::UiExt as _;
+use re_ui::syntax_highlighting::SyntaxHighlightedBuilder;
+
+use super::{FilterError, FilterUiAction, Nullability};
+
+/// A filter for a boolean column.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum BooleanFilter {
+    /// Filter for strictly non-nullable columns.
+    ///
+    /// This includes non-nullable primitive datatypes, as well as outer AND inner non-nullable nested
+    /// datatypes. In that case, the UI should not display the `null` option.
+    NonNullable(bool),
```

Comment:
Does `NonNullable(true)` mean "Show only non-nullable columns?"
What does `NonNullable(false)` then mean?

Reply:
This enum represent UI state. Sometime, we want the UI to offer a `null` choice, and sometime we don't. We make this choice when creating said state, based on the datatype of the column.

So `NonNullable(true)` really means "in the UI don't display a null choice and select the "true" option".